### ISSUE TITLE
Merging a pull request creates a negative issue count

### ISSUE THREAD

**Issue by ciruz** [gogits/gogs#3178]
We DO NOT take questions or config/deploy problems on GitHub, please use our forum: https://discuss.gogs.io

Please take a moment to search that an issue doesn't already exist.

For bug reports, please give the relevant info:

- Gogs version (or commit ref): 0.9.28.0527
- Git version: 
- Operating system:
- Database:
  - [ ] PostgreSQL
  - [X] MySQL
  - [ ] SQLite
- Can you reproduce the bug at http://try.gogs.io:
  - [ ] Yes (provide example URL)
  - [ ] No
  - [ ] Not relevant
- Log gist:

## Description

Whenever I merge a pull request the issues from this repository becomes negative. (Note there where no issues before).

Basically 8 commits and 3 where merged pull requests but now the issues looks like this:
<img src="http://i.imgur.com/P3wm6Wo.png">
...

**Comment by drm** [comment]
+1
Exact same version, also MySQL.

**Comment by BurakDev** [comment]
Duplicate of #3159

**Comment by rudineirk** [comment]
it also happens with postgres

**Comment by tkausl** [comment]
I can not reproduce this on develop-branch

**Comment by roblabla** [comment]
```
MariaDB [gogs]> SELECT num_issues, num_closed_issues, num_pulls, num_closed_pulls FROM repository WHERE id=25;
+------------+-------------------+-----------+------------------+
| num_issues | num_closed_issues | num_pulls | num_closed_pulls |
+------------+-------------------+-----------+------------------+
|         51 |                70 |        67 |               67 |
+------------+-------------------+-----------+------------------+
```

https://github.com/gogits/gogs/blob/e9ae926e040c94ce66666ae6c330c93918fa321c/models/repo.go#L204

`NumOpenIssue (used to calculate the issue count) = num_issues - num_closed_issues`

`num_issues` contains the list of issues that ARE NOT pull request. The problem is, [here](https://github.com/gogits/gogs/commit/e9ae926e040c94ce66666ae6c330c93918fa321c#diff-d4e2553a87eb8b329efd7335f028903cR1833), we set num_closed_issues as all the closed Issues (including PR-issues), which leads to NumOpenIssue being miscalculated.

**Comment by tkausl** [comment]
#3207

**Comment by Unknwon** [comment]
This is fixed in https://github.com/gogits/gogs/pull/3207
